CRM/Logviewer/Page/LogViewer.php

```diff
@@ -31,6 +31,12 @@ public function run() {
       fclose($handle);
       krsort($entries);
       $this->assign('logEntries', $entries);
+
+      $filesize = filesize($logFileName);
+      $sz = 'BKMGTP';
+      $factor = floor((strlen($filesize) - 1) / 3);
+      $filesize = sprintf("%.2f", $filesize / pow(1024, $factor)) . @$sz[$factor];
```

Comment:
This is some pretty cool and scary code to generate the file size in appropriate units! Did you make this up or is it pretty battle tested?

Reply:
Okay, seems like there are a plethora of examples similar to this out on the web and that this should work most of the time. I suppose since this isn't run-time critical code I should be less concerned about edge cases, but for example - what happens if filesize is empty? Then factor is negative one, and all hell breaks loose? Using sz as a string and then getting an index of it is clever but scary for those of us who remember pointer arithmetic and out-of-bounds disasters. So ... I think this is okay except for a test for length of filesize > 0 and sz as an array instead of string.